packages/graph-explorer/src/connector/utils/anySignal.ts

```diff
@@ -4,13 +4,21 @@
  *
  * Requires at least node.js 18.
  *
- * @param signals An array of AbortSignal values that will be merged.
- * @returns A single AbortSignal value.
+ * @param signals A variable amount of AbortSignal values that will be merged in to one.
+ * @returns A single AbortSignal value or undefined if none are passed.
  */
-export function anySignal(signals: AbortSignal[]): AbortSignal {
-  const controller = new AbortController();
+export function anySignal(
+  ...signals: (AbortSignal | null | undefined)[]
+): AbortSignal | undefined {
+  // Filter out null or undefined signals
+  const filteredSignals = signals.flatMap(s => (s ? [s] : []));
```

Comment:
The result is the same at runtime, but it makes a difference to TypeScript.

If I use `.filter()` instead it will complain about the types further down the function.

https://stackoverflow.com/a/59726888/1558906